plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java

```diff
@@ -46,9 +89,88 @@ public String getDescription() {
   }
 
   @Override
-  public IFile createProjectFiles(IProject newProject, AppEngineProjectConfig config, IProgressMonitor monitor)
+  public IFile createAndConfigureProjectContent(IProject newProject, AppEngineProjectConfig config,
+      IProgressMonitor monitor) throws CoreException {
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+    IFile mostImportantFile =  CodeTemplates.materializeAppEngineFlexFiles(newProject, config,
+        subMonitor.newChild(30));
+    configureFacets(newProject, subMonitor.newChild(20));
+    addDependenciesToProject(newProject, subMonitor.newChild(50));
+    return mostImportantFile;
+  }
+
+  /**
+   * Add Java 8 and Dynamic Web Module facet
+   */
+  private void configureFacets(IProject project, IProgressMonitor monitor) throws CoreException {
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+    IFacetedProject facetedProject = ProjectFacetsManager.create(
+        project, true /* convertIfNecessary */, subMonitor.newChild(50));
+    Set<IFacetedProject.Action> facetInstallSet = new HashSet<>();
+    FacetUtil.addJavaFacetToBatch(JavaFacet.VERSION_1_8, facetedProject, facetInstallSet);
+    FacetUtil.addWebFacetToBatch(WebFacetUtils.WEB_30, facetedProject, facetInstallSet);
+    FacetUtil.addFacetSetToProject(facetedProject, facetInstallSet, subMonitor.newChild(50));
+  }
+
+  private void addDependenciesToProject(IProject project, IProgressMonitor monitor)
       throws CoreException {
-    return CodeTemplates.materializeAppEngineFlexFiles(newProject, config, monitor);
+    SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
+
+    // Create a lib folder
+    IFolder libFolder = project.getFolder("lib");
+    if (!libFolder.exists()) {
+      libFolder.create(true, true, subMonitor.newChild(10));
+    }
+
+    // Download the dependencies from maven
+    M2RepositoryService repoService = new M2RepositoryService();
```

Comment:
`M2RepositoryService` is available as an OSGi service through the `ILibraryRepositoryService` service interface, please use that instance instead of creating a new one.